MonoGame.Framework/Graphics/GraphicsResource.cs

```diff
@@ -174,7 +127,23 @@ public GraphicsDevice GraphicsDevice
 
             internal set
             {
+                Debug.Assert(value != null);
+
+                if (graphicsDevice == value)
+                    return;
+
+                // VertexDeclaration objects can be bound to multiple GraphicsDevice objects
+                // during their lifetime. But only one GraphicsDevice should retain ownership.
+                if (graphicsDevice != null)
```

Comment:
This seems fine to me.  It is good that we're protecting against the case of a resource changing graphics devices if it ever happens.  It still is the responsibility of the resource to manage switching of graphics devices if it supports or allows it.